java/kotlin-extractor/BUILD.bazel

```diff
@@ -0,0 +1,181 @@
+"""
+# Usage overview
+Building the extractor can be done via
+```
+bazel build //java/kotlin-extractor:codeql-extractor-kotlin-<variant>-<version>
+```
```

Comment:
It looks like this can't be done from `<root>`, but can be done from `<root>/ql` or `<root>/ql/java/kotlin-extractor`. Is that expected?

Also, I think that this should mention that you needs `tools/bazel` to be the bazel on your path.

Reply:
> It looks like this can't be done from <root>, but can be done from <root>/ql or <root>/ql/java/kotlin-extractor. Is that expected?

I can add a note about building from the internal repo (`bazel build @codeql//java/kotlin-extractor:etc` is what works from there)

> Also, I think that this should mention that you needs tools/bazel to be the bazel on your path.

Actually, any installed bazel or bazelisk will do (bar http fetching problems :sweat_smile:). On `codeql`, if it's bazelisk (which is now the recommended `bazel` executable to have and is what `bazel` is on github actions) it will pick up the correct `bazel` version. If it's a fixed `bazel` version, it will exit any way with a helpful error on a version mismatch.

Furthermore, on `semmle-code`, any installed `bazel` will actually auto-pick `tools/bazel`.

Reply:
Doesn't work for me:
```
$ bazel build //java/kotlin-extractor:codeql-extractor-kotlin-standalone-1.6.0
ERROR: The project you're trying to build requires Bazel 7.1.2 (specified in /home/ian/code/code/ql/.bazelversion), but it wasn't found in /home/ian/local/bazel/inst-7.0.2/lib/bazel/bin.

Bazel binaries for all official releases can be downloaded from here:
  https://github.com/bazelbuild/bazel/releases

You can download the required version directly using this command:
  (cd "/home/ian/local/bazel/inst-7.0.2/lib/bazel/bin" && curl -fLO https://releases.bazel.build/7.1.2/release/bazel-7.1.2-linux-x86_64 && chmod +x bazel-7.1.2-linux-x86_64)
```
It does work from semmle-code, though.

Reply:
The helpful error will have the user install bazel, though, and they will have to update their installation when we bump our bazel version. Wouldn't it be better to just tell them to use the tools/bazel they already have?